cider/function/FunctionLookupEngine.cpp

```diff
@@ -22,39 +22,33 @@
 #include "function/FunctionLookupEngine.h"
 #include "cider/CiderException.h"
 
-#define LOAD_EXTENSION_YAML_AND_INITIALIZE_FUNCTION_LOOK_UP(                          \
-    TYPE_NAME, TYPE_YAML_FILE_PATH, FUNCTION_MAPPING_TYPE)                            \
-  cider::function::substrait::SubstraitExtensionPtr extension_function_ptr =          \
-      cider::function::substrait::SubstraitExtension::loadExtension(                  \
-          {getDataPath() + "/" #TYPE_NAME "/" + "" #TYPE_YAML_FILE_PATH ""});         \
-  cider::function::substrait::SubstraitFunctionMappingsPtr func_mappings =            \
-      std::make_shared<const FUNCTION_MAPPING_TYPE>();                                \
-  scalar_function_look_up_ptr_ =                                                      \
-      std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(    \
-          cider_internal_function_ptr, func_mappings);                                \
-  aggregate_function_look_up_ptr_ =                                                   \
-      std::make_shared<cider::function::substrait::SubstraitAggregateFunctionLookup>( \
-          cider_internal_function_ptr, func_mappings);                                \
-  extension_function_look_up_ptr_ =                                                   \
-      std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(    \
-          extension_function_ptr, func_mappings);
+#define LOAD_EXTENSION_YAML_AND_INITIALIZE_FUNCTION_LOOK_UP(                            \
```

Comment:
I think it's better to convert this macro to a method.

Reply:
Use different function mapping type, macro is better.

Reply:
how about template.

Reply:
Done.